api/routes/profile_page.py

```diff
@@ -45,36 +50,38 @@ async def update_profile(userDetails: ProfileUpdate, current_user: User = Depend
             status_code=400, detail=settings.ERRORS.get("UNKNOWN ERROR"))
 
 
-@profile_route.get("/", response_model=PaginatedUsers)
-def get_all_profile(limit: int = Query(default=50, ge=1, le=100), page: int = Query(default=1, ge=1), db: Session = Depends(get_db)):
-    total_users = db.query(User).count()
-    pages = (total_users - 1) // limit + 1
-    offset = (page - 1) * limit
-    users = db.query(User).order_by(desc(User.created_at)).offset(offset).limit(limit).all()
-
-    links = {
-        "first": f"/api/v1/users/?limit={limit}&page=1",
-        "last": f"/api/v1/users/?limit={limit}&page={pages}",
-        "self": f"/api/v1/users/?limit={limit}&page={page}",
-        "next": None,
-        "prev": None,
-    }
-
-    if page < pages:
-        links["next"] = f"/api/v1/users/?limit={limit}&page={page + 1}"
-
-    if page > 1:
-        links["prev"] = f"/api/v1/users/?limit={limit}&page={page - 1}"
-
-    return PaginatedUsers(
-        users=users,
-        total=total_users,
-        page=page,
-        size=limit,
-        pages=pages,
-        links=links,
-    )
+@profile_route.get("/", response_model=Page[ProfileResponse])
+def get_all_profile(skill: str = Query(None, title="Skill", description="The skill to filter users"), 
+                    stack: str = Query(None, title="Stack", description="The stack to filter users"),
+                    active: Optional[bool] = None, p: Optional[str] = None,
+                    db: Session = Depends(get_db)):
 
+    query = db.query(User)
+    
+    if skill:
+        users_query = query.join(users_skills.UserSkill).join(Skill).filter(Skill.name == skill.capitalize())
+        return paginate(users_query.order_by(desc(User.created_at)))
+    
+    if stack:
+        users_query = query.filter(User.stack.has(name=stack.capitalize()))
+        return paginate(users_query.order_by(desc(User.created_at)))
+    
+    if active is not None:
```

Comment:
In this if-else block, you entirely ignored the base query (query = db.query(User)). I'm not sure if it's an oversight or intentional. Can you clarify? If it's not intentional, I think using the base query here will be better.

Reply:
I guess it is an oversight, I will fix that.